core/src/main/java/com/nisovin/magicspells/Spell.java

```diff
@@ -1247,11 +1307,17 @@ public void setCooldown(final LivingEntity livingEntity, float cooldown, boolean
 			nextCast.remove(uuid);
 			chargesConsumed.remove(uuid);
 		}
+
 		if (serverCooldown > 0)
 			nextCastServer = System.currentTimeMillis() + (long) (serverCooldown * TimeUtil.MILLISECONDS_PER_SECOND);
+
 		if (activateSharedCooldowns && sharedCooldowns != null) {
-			for (Map.Entry<Spell, Float> scd : sharedCooldowns.entrySet()) {
-				scd.getKey().setCooldown(livingEntity, scd.getValue(), false);
+			for (SharedCooldown sharedCooldown : sharedCooldowns) {
+				Float cd = sharedCooldown.cooldown().get(data);
+				if (cd == null) continue;
+
+			    for (Spell spell : sharedCooldown.spells)
```

Comment:
Spaces, but if you're not rebasing for any other reason, there's multiple other cases of this we should fix, so you don't need to force-push just because of this one.